src/features/defi/providers/cosmos/components/CosmosManager/Deposit/CosmosDeposit.tsx

```diff
@@ -0,0 +1,135 @@
+import { useSelector } from 'react-redux'
+import { CircularProgress } from 'components/CircularProgress/CircularProgress'
+import { DefiStepProps, Steps } from 'components/DeFi/components/Steps'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { useBrowserRouter } from 'hooks/useBrowserRouter/useBrowserRouter'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { useCosmosSdkStakingBalances } from 'pages/Defi/hooks/useCosmosSdkStakingBalances'
+import {
+  selectAssetById,
+  selectMarketDataById,
+  selectPortfolioLoading,
+  selectValidatorByAddress,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { Confirm } from './components/Confirm'
+import { Deposit } from './components/Deposit'
+import { Status } from './components/Status'
+import { CosmosDepositActionType } from './DepositCommon'
+import { DepositContext } from './DepositContext'
+import { initialState, reducer } from './DepositReducer'
+
+export const CosmosDeposit = () => {
+  const translate = useTranslate()
+  const [state, dispatch] = useReducer(reducer, initialState)
+  const { query, history, location } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const { chainId, contractAddress, assetReference } = query
+  const assetNamespace = 'slip44'
+  const assetId = toAssetId({ chainId, assetNamespace, assetReference })
+
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const marketData = useAppSelector(state => selectMarketDataById(state, assetId))
+
+  // user info
+  const { state: walletState } = useWallet()
+  const loading = useSelector(selectPortfolioLoading)
+
+  const validatorData = useAppSelector(state => selectValidatorByAddress(state, contractAddress))
+
+  const apr = useMemo(() => bnOrZero(validatorData?.apr).toString(), [validatorData])
+
+  const opportunities = useCosmosSdkStakingBalances({ assetId })
+  const cosmosOpportunity = useMemo(
+    () =>
+      opportunities?.cosmosSdkStakingOpportunities?.find(
+        opportunity => opportunity.address === contractAddress,
+      ),
+    [opportunities, contractAddress],
+  )
+  useEffect(() => {
+    ;(async () => {
+      try {
+        if (!cosmosOpportunity) return
+
+        const chainAdapterManager = getChainAdapterManager()
+        const chainAdapter = chainAdapterManager.get(chainId)
+        if (!(walletState.wallet && contractAddress && chainAdapter && apr)) return
+
+        const address = await chainAdapter.getAddress({ wallet: walletState.wallet })
+
+        dispatch({ type: CosmosDepositActionType.SET_USER_ADDRESS, payload: address })
+        dispatch({
+          type: CosmosDepositActionType.SET_OPPORTUNITY,
+          payload: { ...cosmosOpportunity },
+        })
+      } catch (error) {
+        // TODO: handle client side errors
+        console.error('CosmosDeposit error:', error)
```

Comment:
moduleLogger

Reply:
Also from the Foxy implementation, will add it in the same stacked PR

Reply:
https://github.com/shapeshift/web/pull/2323